### PR TITLE
Filter out empty task descriptions

### PR DESCRIPTION
**Pull request by amitaibu**
Currently when editing an existing item and clearing the text, `onEnter` and `onBlur` we get empty lines. 

elm_ _todomvc

PR adds a `SubmitTask` action, and filters out empty descriptions.

(btw, it's my first elm PR! :smile: )

### CODE REVIEW

File: Todo.elm [hunk 1]
```diff
@@ -110,6 +111,14 @@ update action model =
           in
               { model | tasks <- List.map updateTask model.tasks }
 
+      {- UpdateTask updates the description, but we want to know when it was
+      actually submitted, so we can filter out empty descriptions. -}
+      SubmitTask id ->
+          let
+              newModel = update (EditingTask id False) model
+          in
+              { newModel | tasks <- List.filter (\t -> String.length t.description > 0) newModel.tasks }
```
Comment (passy) [inline Todo.elm]: Could you throw a `String.trim` into the mix here? The official TodoMVC guidelines require that just whitespaces are treated as empty, too. :)

Comment (amitaibu) [inline Todo.elm]: :+1: 
I've actually just noted thanks to you comment, the it is possible in the current master to submit a line with empty spaces, so I'll try to take care of it as-well ;)

Comment (passy) [inline Todo.elm]: Fantastic! :+1:

**Comment by amitaibu** [comment]
@passy PR now also checks if the added string is valid to begin with. Ready for re-review.

File: Todo.elm [hunk 2]
```diff
@@ -129,6 +138,10 @@ update action model =
       ChangeVisibility visibility ->
           { model | visibility <- visibility }
 
+-- Detrmine if a String is valid and should be added.
+isValidString : String -> Bool
+isValidString string =
+    String.length (String.trim string) > 0
```
Comment (passy) [inline Todo.elm]: I think you could use `String.isEmpty` here. :)

Comment (amitaibu) [inline Todo.elm]: will do tomorrow from the office, stay tuned ;)

**Comment by amitaibu** [comment]
@passy using `String.isEmpty` now, thanks.

**Comment by evancz** [comment]
This is fixed in the latest version afaik

**Pull request closed without merging**